app/main.py

```diff
@@ -1,34 +1,52 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple, end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        if self.start[0] == self.end[0]:
+            self.decks = [
+                Deck(self.start[0], i)
+                for i in range(self.start[1], self.end[1] + 1)
+            ]
+        elif self.start[1] == self.end[1]:
+            self.decks = [
+                Deck(i, self.start[0])
+                for i in range(self.start[0], self.end[0] + 1)
+            ]
```

Comment:
For me it’s looks too heavy for a constructor. I would prefer to separate this logic to a method or do some refactor to make it looks a bit compact and readable